packages/transformers/js/core/src/collect.rs

```diff
@@ -681,6 +693,10 @@ impl Visit for Collect {
       Expr::This(_this) => {
         if self.in_module_this {
           handle_export!();
+        } else if !self.in_class {
+          if let MemberProp::Ident(prop) = &node.prop {
+            self.this_exprs.insert(id!(prop), node.clone());
+          }
```

Comment:
Do we only want to handle `this` in exported functions? If you have a non-exported function that uses `this` we shouldn't need to wrap. You could track which functions use `this` instead of each property access, and then later check if any of the exported functions used `this`. Might be more accurate than just looking for property names?

Reply:
If tracking functions that use this, we would have to keep some sort of struct that maps the Ident for the original exported function to every re-assignment of the function to another variable. Discussed this with @imbrian and we were thinking that the overhead from wrapping every time a `this` is encountered seems better than having to keep a (possibly very large) struct of every possible alias referring back to a function that calls `this`.